swift/ql/src/queries/Security/CWE-321/HardcodedEncryptionKey.ql

```diff
@@ -0,0 +1,70 @@
+/**
+ * @name Hard-coded encryption key
+ * @description Using hardcoded keys for encryption is not secure, because potential attacker can easiy guess them.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 8.1
+ * @precision high
+ * @id swift/hardcoded-key
+ * @tags security
+ *       external/cwe/cwe-321
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import DataFlow::PathGraph
+
+/**
+ * An `Expr` that is used to initialize a key.
+ */
+abstract class KeySource extends Expr { }
+
+/**
+ * The creation of a literal byte array.
+ */
+class ByteArrayLiteralSource extends KeySource {
+  ByteArrayLiteralSource() { this = any(ArrayExpr arr | arr.getType().toString() = "Array<UInt8>") }
+}
+
+/**
+ * Any string literal as a key source.
+ */
+class StringLiteralSource extends KeySource {
+  StringLiteralSource() { this instanceof StringLiteralExpr }
+}
```

Comment:
If you want to be fancy:
```suggestion
class StringLiteralSource extends KeySource instanceof StringLiteralExpr { }
```
You can read about this use of `instanceof` [here](https://codeql.github.com/docs/ql-language-reference/types/#non-extending-subtypes).

Reply:
Neat! Changed it to the less verbose syntax that you suggested.